src/main/java/view/SearchHistoryPanel.java

```diff
@@ -0,0 +1,73 @@
+package view;
+
+import interface_adapter.search.SearchHistoryController;
+import app.Session;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.List;
+import org.json.JSONObject;
+
+public class SearchHistoryPanel extends JPanel {
+    public SearchHistoryPanel(SearchHistoryController controller, MainFrame frame) {
+        setLayout(new BorderLayout());
+
+        // Search History Display
+        JTextArea historyArea = new JTextArea(15, 40);
+        historyArea.setEditable(false);
+        JScrollPane scrollPane = new JScrollPane(historyArea);
+
+        // Top Panel: Back Button and Show History Button
+        JPanel topPanel = new JPanel(new FlowLayout(FlowLayout.LEFT));
+        JButton backButton = new JButton("Back");
+        backButton.addActionListener(e -> frame.switchTo("Home")); // Switch back to HomePanel
+        topPanel.add(backButton);
+
+        JButton showHistoryButton = new JButton("Show History");
+        showHistoryButton.addActionListener(e -> updateHistory(historyArea));
```

Comment:
Confirm that updateHistory() is optimized for large datasets. If the history list grows, consider paginating or limiting the display.